test/e2e/unmanaged_dns_test.go

```diff
@@ -240,8 +263,35 @@ func TestUnmanagedDNSToManagedDNSInternalIngressController(t *testing.T) {
 		t.Fatalf("failed to update ingresscontroller %s: %v", name, err)
 	}
 
-	if err := kclient.Delete(context.TODO(), lbService); err != nil && !errors.IsNotFound(err) {
-		t.Fatalf("failed to delete svc %s: %v", lbService.Name, err)
+	var oldLoadBalancerStatus corev1.LoadBalancerStatus
+	lbService.Status.LoadBalancer.DeepCopyInto(&oldLoadBalancerStatus)
+
+	// Only delete the service on platforms that don't automatically update the service's scope.
+	switch platform {
+	case configv1.AlibabaCloudPlatformType, configv1.AWSPlatformType, configv1.IBMCloudPlatformType, configv1.PowerVSPlatformType:
+		if err := kclient.Delete(context.TODO(), lbService); err != nil && !errors.IsNotFound(err) {
+			t.Fatalf("failed to delete svc %s: %v", lbService.Name, err)
+		}
+	}
+
+	// Ensure the service's load-balancer status changes.
+	err := wait.PollImmediate(10*time.Second, 5*time.Minute, func() (bool, error) {
+		lbService := &corev1.Service{}
+		if err := kclient.Get(context.TODO(), controller.LoadBalancerServiceName(ic), lbService); err != nil {
+			t.Logf("Get %q failed: %v, retrying ...", controller.LoadBalancerServiceName(ic), err)
+			return false, nil
+		}
+		if reflect.DeepEqual(lbService.Status.LoadBalancer, oldLoadBalancerStatus) {
+			t.Logf("Waiting for service %q to be updated", controller.LoadBalancerServiceName(ic))
+			return false, nil
+		} else if ingresscontroller.IsServiceInternal(lbService) {
```

Comment:
Moving this check seems odd - can it magically flip between internal and external?

Reply:
This check was moved just because we needed an up-to-date version of `lbService`. It's in the loop since we have an up-to-date `lbService`. Flipping isn't a concern here, if it detects the lbService was updated, then it checks if it's still internal only once then exits (it's not checking internal/external repeatedly).

This could be moved outside the loop, but just requires another `kclient.Get` prior to checking, so I just threw it in the loop to be more efficient. If you feel being outside of the loop would be clearer, let me know.